File: libs/eave-stdlib-ts/src/core-api/operations/github.ts
import {
  RequestArgsOrigin,
  RequestArgsTeamId,
  makeRequest,
} from "../../requests.js";
import {
  GithubInstallation,
  GithubInstallationInput,
} from "../models/github.js";
import { Team } from "../models/team.js";
import { CoreApiEndpointConfiguration } from "./shared.js";

export type GetGithubInstallationRequestBody = {
  github_integration: GithubInstallationInput;
};

export type GetGithubInstallationResponseBody = {
  team: Team | null;
  github_integration: GithubInstallation;
};

export class GetGithubInstallationOperation {
  static config = new CoreApiEndpointConfiguration({
    path: "/integrations/github/query",
  });

  static async perform(
    args: RequestArgsOrigin & { input: GetGithubInstallationRequestBody },
  ): Promise<GetGithubInstallationResponseBody> {
    const resp = await makeRequest({
      config: this.config,
      ...args,
    });
    const responseData = <GetGithubInstallationResponseBody>await resp.json();
    return responseData;
  }
}

export type DeleteGithubInstallationRequestBody = {
  github_integration: GithubInstallationInput;
};

export class DeleteGithubInstallationOperation {
  static config = new CoreApiEndpointConfiguration({
    path: "/integrations/github/delete",
  });

  static async perform(
    args: RequestArgsTeamId & { input: DeleteGithubInstallationRequestBody },
  ): Promise<void> {
    await makeRequest({
      config: this.config,
      ...args,
    });
  }
}

File: libs/eave-stdlib-py/src/eave/stdlib/core_api/operations/github.py
from typing import Optional, Unpack
import uuid
from eave.stdlib.core_api.models.github import GithubInstallation
from eave.stdlib.core_api.models.github import GithubInstallationInput
from . import BaseRequestBody, BaseResponseBody, CoreApiEndpoint, CoreApiEndpointConfiguration

from ..models import team
from ... import requests


class GetGithubInstallation(CoreApiEndpoint):
    config = CoreApiEndpointConfiguration(
        path="/integrations/github/query",
        auth_required=False,
        team_id_required=False,
    )

    class RequestBody(BaseRequestBody):
        github_integration: GithubInstallationInput

    class ResponseBody(BaseResponseBody):
        team: Optional[team.Team]
        github_integration: GithubInstallation

    @classmethod
    async def perform(
        cls,
        input: RequestBody,
        **kwargs: Unpack[requests.CommonRequestArgs],
    ) -> ResponseBody:
        response = await requests.make_request(
            config=cls.config,
            input=input,
            **kwargs,
        )

        body = await cls.make_response(response, cls.ResponseBody)
        return body


class DeleteGithubInstallation(CoreApiEndpoint):
    config = CoreApiEndpointConfiguration(
        path="/integrations/github/delete",
        auth_required=False,
    )

    class RequestBody(BaseRequestBody):
        github_integration: GithubInstallationInput

    class ResponseBody(BaseResponseBody):
        pass

    @classmethod
    async def perform(
        cls,
        input: RequestBody,
        team_id: uuid.UUID,
        **kwargs: Unpack[requests.CommonRequestArgs],
    ) -> ResponseBody:
        response = await requests.make_request(
            config=cls.config,
            input=input,
            team_id=team_id,
            **kwargs,
        )

        body = await cls.make_response(response, cls.ResponseBody)
        return body
